src/figuresClasses.ts

```diff
@@ -1,19 +1,98 @@
-export interface Figure {
+// import { type } from "os";
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('the entered value must be greater than 0');
+    }
+
+    if (this.a >= this.b + this.c) {
+      throw new Error(
+        'side "a" must be less than the sum of the other two sides',
+      );
+    }
+
+    if (this.c >= this.b + this.a) {
+      throw new Error(
+        'side "c" must be less than the sum of the other two sides',
+      );
+    }
+
+    if (this.b >= this.a + this.c) {
+      throw new Error(
+        'side "b" must be less than the sum of the other two sides',
+      );
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
 
+    const area: number = (Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c),
+    )
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?

Reply:
Why didn’t you create a separate function for rounds down to hundredths? You duplicate code